### ISSUE TITLE
Minify CSS, HTML and JS before merging with Main branch

### ISSUE THREAD

**Issue by OscarLarm** [annk15/MTGKeywordHelper#11]
Remember to minify the frontend parts into a optimized branch before merging into the main branch for slight performance gain!

**Comment by annk15** [comment]
I think we should focus on keeping the readability in the main branch also as we may need to clone from this when fixing bugs in the future but we could minify the the frontend code before we create a release.

**Comment by OscarLarm** [comment]
> I think we should focus on keeping the readability in the main branch also as we may need to clone from this when fixing bugs in the future but we could minify the the frontend code before we create a release.


Yeah waiting until the release build to minify is definetly smarter. Something we could do is automate minify to happen automatically when the working css file is saved, and save the minify version in a seperate release-directory. We could do this with Node.js and Gulp.

**Comment by annk15** [comment]
> > I think we should focus on keeping the readability in the main branch also as we may need to clone from this when fixing bugs in the future but we could minify the the frontend code before we create a release.
> 
> 
> 
> 
> 
> Yeah waiting until the release build to minify is definetly smarter. Something we could do is automate minify to happen automatically when the working css file is saved, and save the minify version in a seperate release-directory. We could do this with Node.js and Gulp. 

Yes that sounds good, if it slow down the ide too much we could create do the optimisation automatically when the dev branch is pushed in to master. We will later tag the master branch with release tags when we feel happy with the state.
